ml/Host.cc

```diff
@@ -477,11 +482,13 @@ void DetectableHost::detect() {
     heartbeat_init(&HB);
 
     while (!netdata_exit) {
+        netdata_thread_testcancel();
```

Comment:
The only place that really needs this check is https://github.com/netdata/netdata/pull/12681/files#diff-de8b578e59a869a2e6caf304130d076fd5971e2793c93e0fe00a73df46126cc7R379, because otherwise we end up calling sleep which is a cancellation point.

Regardless, I added an explicit cancellation test just to be on the safe side for the upcoming release.